src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -554,13 +562,19 @@ private async Task LoadAsync(bool ignoreFailures, CancellationToken cancellation
 
             var startupExceptions = new List<Exception>();
 
+            IAsyncEnumerator<ConfigurationClient> clientEnumerator = null;
+
             try
             {
                 while (true)
                 {
-                    IEnumerable<ConfigurationClient> clients = _configClientManager.GetAllClients();
+                    clientEnumerator = _configClientManager.GetAllClients(cancellationToken).GetAsyncEnumerator();
 
-                    if (await TryInitializeAsync(clients, startupExceptions, cancellationToken).ConfigureAwait(false))
+                    await using ConfiguredAsyncDisposable configuredAsyncDisposable = clientEnumerator.ConfigureAwait(false);
+                    // Guaranteed to have at least one available client since it is a application startup path.
+                    _ = await clientEnumerator.MoveNextAsync().ConfigureAwait(false);
+
+                    if (await TryInitializeAsync(clientEnumerator, startupExceptions, cancellationToken).ConfigureAwait(false))
```

Comment:
Passing enumerator as argument is risky. Subsequent iterations will modify/invalidate state. Something that should be avoided for [in] parameters.